Use direct file storage for ActivityPub inbox attachments

obenland opened the pull request:
## Summary
Implements a dual storage strategy where Mastodon imports continue using WordPress attachment posts while ActivityPub inbox items use direct file storage in `uploads/activitypub/{post_id}/`.

## Changes

### New methods added:
- **`import_files()`**: Public method for direct file import (used by ap_posts)
- **`save_file()`**: Downloads and saves files to `uploads/activitypub/{post_id}/`
  - Returns array with `url`, `mime_type`, and `alt` keys
- **`import_inline_files()`**: Processes inline images for file storage
- **`append_files_to_content()`**: Appends file-based media to content
- **`generate_files_markup()`**: Creates block markup from file data
- **`get_files_gallery_block()`**: Gallery block for file-based attachments
- **`delete_files_directory()`**: Public helper to delete post's file directory

### Existing methods preserved:
- **`import()`**: Still used for Mastodon imports (creates attachment posts)
- All attachment-related helper methods remain unchanged

## Key improvements
- Uses WP_Filesystem API exclusively for all file operations
- Only handles remote URLs (no local file logic needed for inbox items)
- Reusable `delete_files_directory()` helper used in multiple places
- Removed unused function parameters (`author_id`, `path` key)
- Well-documented return types with hash notation for array keys
- Comprehensive parameter documentation for file data arrays
- Simple folder-per-post structure for easy cleanup

## Benefits
- ✅ No attachment posts for inbox items (cleaner database)
- ✅ Clear separation between import types
- ✅ Consistent use of WordPress filesystem abstractions
- ✅ Mastodon imports unaffected (backward compatible)

## Test plan
- [x] All 1212 tests passing
- [x] Tests updated to verify file creation instead of attachment posts
- [x] Test cleanup properly removes activitypub directories
- [x] Mastodon import tests still pass (no regression)

## Technical details
The implementation uses a simple folder-per-post structure where each `ap_post` gets its own directory at `uploads/activitypub/{post_id}/`. When a post is deleted, the entire directory is recursively removed using `WP_Filesystem::delete()`.

Files are tracked by the directory structure itself rather than post meta, keeping the implementation simple and avoiding the need for complex deduplication logic.

obenland committed:
Add direct file storage for ActivityPub inbox attachments

Implements a dual storage strategy where Mastodon imports continue using
WordPress attachment posts while ActivityPub inbox items use direct file
storage in uploads/activitypub/{post_id}/.

New methods added:
- import_files(): Public method for direct file import (used by ap_posts)
- save_file(): Downloads and saves files to uploads/activitypub/{post_id}/
  Returns array with 'url', 'mime_type', and 'alt' keys
- import_inline_files(): Processes inline images for file storage
- append_files_to_content(): Appends file-based media to content
- generate_files_markup(): Creates block markup from file data
- get_files_gallery_block(): Gallery block for file-based attachments
- delete_files_directory(): Public helper to delete post's file directory

Existing methods preserved:
- import(): Still used for Mastodon imports (creates attachment posts)
- All attachment-related helper methods remain unchanged

Key improvements:
- Uses WP_Filesystem API exclusively for all file operations
- Only handles remote URLs (no local file logic needed for inbox items)
- Reusable delete_files_directory() helper used in multiple places
- Removed unused function parameters (author_id, path key)
- Well-documented return types with hash notation for array keys
- Comprehensive parameter documentation for file data arrays
- Simple folder-per-post structure for easy cleanup

Benefits:
- No attachment posts for inbox items (cleaner database)
- Clear separation between import types
- Consistent use of WordPress filesystem abstractions
- Mastodon imports unaffected (backward compatible)

Tests updated to verify file creation and cleanup.
All 1212 tests passing.

obenland committed:
Simplify separator assignment in post update

Refactored the logic for assigning the separator before updating post content by using a ternary operator instead of an if statement. This improves code readability and conciseness.

obenland committed:
Fix code quality feedback.

obenland committed:
Use built-in function

pfefferle commented on includes/class-attachments.php:
```diff
@@ -32,8 +32,22 @@ public static function delete_attachments_with_post( $post_id ) {
+		\WP_Filesystem();
+		global $wp_filesystem;
+
+		$activitypub_dir = \wp_upload_dir()['basedir'] . '/activitypub/' . $post_id;
```
when we add a subpath or a prefix, we can re-use it for avatars, emojis and comments!

obenland replied:
As in, we should make the path `/activitypub/ap_post/` or something?

pfefferle replied:
yes

obenland replied:
Fixed in [c49d1d0](https://github.com/Automattic/wordpress-activitypub/pull/2375/commits/c49d1d053835f860bbc6f68cff0779737c778afc).

pfefferle commented on includes/collection/class-posts.php:
```diff
@@ -135,13 +135,9 @@ public static function update( $activity, $user_id ) {
-			// Add new attachments.
-			Attachments::import( $activity['object']['attachment'], $post_id );
+			// Delete old files and add new attachments.
+			Attachments::delete_files_directory( $post_id );
```
```suggestion
			Attachments::delete_directory( $post_id );
```

obenland replied:
Fixed in [998faba](https://github.com/Automattic/wordpress-activitypub/pull/2375/commits/998faba4300ee79eee1db3c72c9aebefc607832a).

pfefferle commented on includes/class-attachments.php:
```diff
@@ -32,8 +32,22 @@ public static function delete_attachments_with_post( $post_id ) {
 
-		foreach ( \get_attached_media( '', $post_id ) as $attachment ) {
-			\wp_delete_attachment( $attachment->ID, true );
+		self::delete_files_directory( $post_id );
```
```suggestion
		self::delete_directory( $post_id );
```

obenland replied:
Fixed in [998faba](https://github.com/Automattic/wordpress-activitypub/pull/2375/commits/998faba4300ee79eee1db3c72c9aebefc607832a).

obenland committed:
Rename delete_files_directory to delete_directory

Refactored Attachments class to rename the delete_files_directory method to delete_directory for clarity and consistency. Updated all references to the method in related files.

obenland committed:
Update media storage path for ap_post attachments

Changed the directory for storing ap_post media files from 'uploads/activitypub/{post_id}/' to 'uploads/activitypub/ap_posts/{post_id}/'. Introduced a static property for the directory path and updated all relevant references to improve organization and clarity.
